### ISSUE TITLE
[pip] no support for specifying the python version to use for a virtualenv?

### ISSUE THREAD

**Issue by wbolster** [ansible/ansible-modules-core#586]
It seems the `pip` module doesn't allow me to specify which Python version to use for a virtualenv.

When I use something like this:

```
pip: name=myproject virtualenv=/path/to/virtualenv/
```

... the system Python version is used, which is Python 2.7 on my system. However, my project requires Python 3, and Python 3.4 is installed properly on the system. The problem is that there doesn't seem to be a way to tell the `pip` module to create the virtualenv using Python 3.4.

Fwiw, when manually creating a virtualenv, I would use `virtualenv -ppython3.4 myvenv` to create it.

**Comment by cchurch** [comment]
One approach you could use would be to create the virtualenv first and specify the Python version:

```
command: creates=/path/to/virtual virtualenv -p python3.4 /path/to/virtualenv/
```

Then, since the virtualenv already exists, the pip module won't try to create it and _should_ use the version of Python in the virtualenv.

**Comment by wbolster** [comment]
Thanks, I had a work-around like that in place already.

However, I still think this needs to be in Ansible's `pip` module. virtualenvs are intended for isolated python environments, and that includes not only python packages, but also the python version used.

Proposal:  add a `virtualenv_python` (or maybe `python_version`?) argument.

**Comment by wbolster** [comment]
Just to be clear: some systems have separate `virtualenv-2.7` and `virtualenv-3.4` commands that may do the right thing, but these are not guaranteed to be installed, since it requires multiple `virtualenv` installations, and distros may ship only a single version. Since `virtualenv` is designed to create environments for different Python versions (using the `-p` flag), this is perfectly fine. It just means that Ansible's current `virtualenv_command` does not cover all situations, hence the `virtualenv_python` argument I've added in PR #629, which, together with `virtualenv_command`, does cover all situations.

**Comment by bivald** [comment]
This would help me as well, specially since it appears as newer Debian (which can apt-get python3) doesn't get virtualenv-3.4 link, just virtualenv.

**Comment by wbolster** [comment]
Unfortunately my PR #629 has been lying around for months, slowly gathering dust and bitrot. :-(

**Comment by bivald** [comment]
I would assume they are working on it, but that they simply have quite the backlog.

**Comment by glassresistor** [comment]
this would help me as well. the upside of ansible is cleanly specifying something, having to use two tasks makes it less clean.

**Comment by lacion** [comment]
in a newly freshly installed vagrant box with ubuntu 14.04 using virtualenv-3.4 results in a virtualenv with python 2.7.

using virtualenv-3.4 its only running virtualenv with python3 but virtualenv is creating the env with python witch defaults to python2.7

**Comment by wbolster** [comment]
fixed via #629
